src/files/ThemeXml.ts

```diff
@@ -0,0 +1,149 @@
+ * determining which fonts to apply when the StylesXml cannot determine which
+ * font to apply for the required Normal style in MS Word.
+ *
+ * @TODO Implement ColorScheme and FormatScheme
+ */
+export type ThemeElements = {
+	fontScheme: FontScheme;
+}
+
+export type FontScheme = {
+	name: string;
+	majorFont: {
+		latinFont: LatinFont;
+		otherFonts: Font[];
+	};
+	minorFont: {
+		latinFont: LatinFont;
+		otherFonts: Font[]
+	}
+}
+
+export type Font = {
+	script?: string;
+	typeface: string;
+}
+
+export interface LatinFont extends Font {
+	panose: string;
+}
+
+export class ThemeXml extends XmlFile {
+	public static contentType = FileMime.theme;
+	public readonly themeElements: ThemeElements;
+
+	public constructor(location: string) {
+		super(location);
+		const newThemeElements = {} as ThemeElements;
+		this.themeElements = newThemeElements;
+	}
+
+	public toNode(): Document {
+		return create(
+			`<a:theme xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main">
+			{
+				element a:themeElements {
+					element a:fontScheme {
+						attribute name { $fontSchemeName },
+						element a:majorFont {
+							element a:latin {
+								attribute typeface { $majorFontLatinTypeface },
+								attribute panose { $majorFontLatinPanose }
+							},
+							for $font in array:flatten($majorOtherFonts)
+							return element a:font {
+								attribute script { $font('script') },
+								attribute typeface { $font('typeface') }
+							}
+						},
+						element a:minorFont {
+							element a:latin {
+								attribute typeface { $minorFontLatinTypeface },
+								attribute panose { $minorFontLatinPanose }
+							},
+							for $font in array:flatten($minorOtherFonts)
+							return element a:font {
+								attribute script { $font('script') },
+								attribute typeface { $font('typeface') }
+							}
+						}
+					}
+				}
+			}</a:theme>`,
+			{
+				fontSchemeName: this.themeElements.fontScheme.name,
+				majorFontLatinTypeface: this.themeElements.fontScheme.majorFont.latinFont.typeface,
+				majorFontLatinPanose: this.themeElements.fontScheme.majorFont.latinFont.panose,
+				majorOtherFonts: this.themeElements.fontScheme.majorFont.otherFonts,
+				minorFontLatinTypeface: this.themeElements.fontScheme.minorFont.latinFont.typeface,
+				minorFontLatinPanose: this.themeElements.fontScheme.minorFont.latinFont.panose,
+				minorOtherFonts: this.themeElements.fontScheme.minorFont.otherFonts
+			},
+			true
+		);
+	}
+
+	/**
+	 * Instantiate this class by looking at the DOCX XML for it.
+	 */
+	public static async fromArchive(archive: Archive, location?: string, xml?: Document): Promise<ThemeXml> {
+		// If a location is supplied, use that, otherwise use the default location for theme files.
+		location = location ?? 'word/theme/theme1.xml';
+		const themeDocument = archive.hasFile(location) ? await archive.readXml(location) : xml;
+		const fontScheme: Record<string, string | Font[]> = evaluateXPathToMap(`
```

Comment:
This type could be much stricter/more accurate, for some properties the query you've written already tells you if something is gonna be an `array{}`, or possibly a sequence of strings or null.

`fontoxpath` does some magic around the `./derp/string()` query because "a sequence of strings" doesnt really exist in JS; fontoxpath may return a `string`, `string[]` or `null`. You may need to get a little more familiar with that, or rewrite your queries as `string(./derp)` or `array { ./derp/string() }` to better predict all realistic outcomes or its typing.

You _might_ even improve the query itself to the point where you don't need to further massage this data in Javascript in lines 116-143 👀